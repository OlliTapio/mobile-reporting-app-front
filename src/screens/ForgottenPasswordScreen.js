import React from 'react';
import { View, Text, TouchableHighlight } from 'react-native';
import forgottenPasswordStyles from './style/forgottenPasswordStyles';

export default class ForgottenPasswordScreen extends React.Component {

  handleButtonPress = () => {
      this.props.navigation.goBack();
  };

  render() {
      return (
          <View style={forgottenPasswordStyles.container}>
              <TouchableHighlight style={forgottenPasswordStyles.button} onPress={() => this.handleButtonPress()}>
                  <Text style={forgottenPasswordStyles.text}>Forgotten Password Screen</Text>
              </TouchableHighlight>
          </View>
      );
  }
}