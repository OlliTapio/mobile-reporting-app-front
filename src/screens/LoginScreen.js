import React from 'react';
import { Text, ImageBackground } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';
import { SignInButton } from '../components/Button';
import { Input } from '../components/TextInput';

export default class LoginScreen extends React.Component {
    static propTypes = {
        navigation: PropTypes.object,
    };

    constructor(props)
    {
        super(props);
        this.state = {
            isLoading: true,
            TextInputUser: '',
            TextInputPassword: '',
            TextInputServer: ''
        };
    }

    render() {
        return (
            <ImageBackground source={require('./images/background.png')} style={styles.contentContainer}>
                <Text style={styles.title}>
                    MR-Application
                </Text>

                <Text style={styles.slogan}>
                    Keep calm and keep reporting
                </Text>

                <Input
                    source={require('./images/person.png')}
                    placeholder='Email'
                    placeholderTextColor='white'
                    onChangeText={TextInputUser => this.setState({ TextInputUser })}
                />
                <Input
                    source={require('./images/lock.png')}
                    secureTextEntry={true}
                    placeholder='Password'
                    placeholderTextColor='white'
                    onChangeText={TextInputPassword => this.setState({ TextInputPassword })}
                />
                <Input
                    source={require('./images/language.png')}
                    placeholder='Server url'
                    placeholderTextColor='white'
                    onChangeText={TextInputServer => this.setState({ TextInputServer })}
                />

                <SignInButton onPress={() => this.props.navigation.navigate('drawerStack')}>
                    Sign In
                </SignInButton>

                <Text style={styles.textLink} onPress={() => this.props.navigation.navigate('forgottenPasswordScreen')}>
                    Forgot password?
                </Text>

                <Text style={styles.signUp} onPress={() => this.props.navigation.navigate('signUpScreen')}>
                    Sign up
                </Text>

                <Text style={styles.copyright}>
                    Copyright Â© Arter Oy 2017
                </Text>
            </ImageBackground>
        );
    }
}