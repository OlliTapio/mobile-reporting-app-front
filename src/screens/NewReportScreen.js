import React from 'react';
import { StyleSheet, View, TextInput, Platform, Alert, Button } from 'react-native';
import { url } from './urlsetting';
import { NavigationActions } from 'react-navigation';
export default class NewFormScreen extends React.Component {


    constructor(props)
    {
        super(props);
        this.state = {
            TextInputName  : '',    // Text input is initialized as an empty string.
            templateID     : '',    /* TemplateID that is inherited from navigation parameters
                                       as stated in TemplateScreen class. */
        };

    }

    // Gets the current date and returns it as a string.

    getDate = () => {

        const today = new Date();
        let dd = today.getDate();
        let mm = today.getMonth() + 1; // January is 0!
        const yyyy = today.getFullYear();

        if (dd < 10) {
            dd = '0' + dd;
        }

        if (mm < 10) {
            mm = '0' + mm;
        }

        return yyyy + '-' + mm + '-' + dd;
    };


    // Inserts data to server with a post method.

    InsertDataToServer = () => {

        const date = this.getDate();

        fetch(url + '/users/1/forms', {     // ***NOTE*** Change to /reports when API has been changed too.
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },

            body: JSON.stringify({
                templateID: this.props.navigation.state.params.templateID,
                title: this.state.TextInputName,
                dateCreated: date,
                answers: [
                    {
                        fieldID: 1,
                        answer: 'Answer 1'
                    },
                    {
                        fieldID: 2,
                        answer: 'Answer 2'
                    }
                ]

            })

        }).then(response => {
            if (response.status === 200) {
                this.props.navigation.state.params.refresh();
                this.props.navigation.dispatch(NavigationActions.back());
                return 'Report added';
            } else {
                return response.status;
            }
        }).then((message) => {
            // Showing response message coming from server after inserting records.
            Alert.alert(message);

        }).catch((error) => {
            console.error(error);
        });

    };

    render() {
        return (
            <View style={styles.MainContainer}>
                <TextInput
                    placeholder='Enter Report Name'

                    onChangeText={TextInputName => this.setState({ TextInputName })}

                    underlineColorAndroid='transparent'

                    style={styles.TextInputStyleClass}
                />
                <Button
                    title='Create new report'
                    onPress={ this.InsertDataToServer }
                >
                </Button>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    MainContainer: {
        justifyContent: 'center',
        flex: 1,
        paddingTop: (Platform.OS === 'ios') ? 20 : 0,
    },

    TextInputStyleClass: {
        textAlign: 'center',
        marginBottom: 7,
        height: 40,
        borderWidth: 1,
        borderColor: '#FF5722',
    },
});