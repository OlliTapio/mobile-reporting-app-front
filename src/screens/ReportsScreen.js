import React, { Component } from 'react';
import {
    View,
    Text,
    ScrollView,
    FlatList,
} from 'react-native';

import { ListItem, SearchBar } from 'react-native-elements';
import Icon from 'react-native-vector-icons/FontAwesome';
import styles from './style/reportsScreenStyle';

class ReportsScreen extends Component {
    constructor(props) {
        super(props);
        this.state = {
            reports: props.navigation.state.params.reports,     // The reports inherited from the layout component in TemplateScreen.
            layoutID: props.navigation.state.params.layoutID,   // The specific layoutID inherited from the layout component in TemplateScreen.
            title: props.navigation.state.params.title,         // The title inherited from the layout component in TemplateScreen.
            nofForms: props.navigation.state.params.nofForms,   // The number of forms inherited from the layout component in TemplateScreen.
        };
    }


    /*
     Calls the inherited createNew function which is explained in TemplateScreen class.
     */
    createNew(layoutID) {
        this.props.navigation.state.params.new(layoutID);
    }

    render() {
        /*
         Renders the reports of a template in a FlatList component.
         */

        return (
            <View style={{ flex: 1 }}>
                <ScrollView style={styles.MainContainer}>
                    <View style={styles.titleContainer}>
                        <Text style={styles.title}>
                            {this.state.title}
                        </Text>
                        <Icon
                            name={'plus'}
                            color={'green'}
                            size={20}
                            onPress={() => this.createNew(this.state.layoutID)} // Navigates to NewReportScreen when pressed.
                        />
                    </View>
                    <View style={styles.section}>
                        <Icon name={'sort'} color={'gray'} size={14}> A-Z</Icon>
                        <SearchBar
                            placeholder={'Search for reports'}
                            lightTheme
                            containerStyle={styles.searchBarContainer}
                            inputStyle={{ backgroundColor: '#fff', width: 160, }}

                        />
                        <Text style={{ fontWeight: 'bold' }}>
                            {this.state.nofForms + ' Forms'}
                        </Text>
                    </View>

                    <View style={styles.container}>
                        <FlatList
                            style={styles.flatList}
                            data={ this.state.reports } // The data in which the reports are stored.
                            renderItem={({ item }) =>   // Renders the reports
                                <ListItem
                                    key={item.title}    // Defines the key of the report to be the title.
                                    title={item.title}  // Title of the report
                                    subtitle={item.dateCreated} // The creation date of the report as a subtitle.
                                    containerStyle={styles.ListItemStyle}
                                />}
                            keyExtractor={item => item.title}
                        />
                    </View>

                </ScrollView>
            </View>
        );
    }
}

export default ReportsScreen;