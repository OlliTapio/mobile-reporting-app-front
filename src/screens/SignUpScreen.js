import React from 'react';
import { View, Text, TouchableHighlight } from 'react-native';
import signUpStyles from './style/signUpStyles';

export default class SignUpScreen extends React.Component {

    handleButtonPress = () => {
        this.props.navigation.goBack();
    };

    render() {
        return (
            <View style={signUpStyles.container}>
                <TouchableHighlight style={signUpStyles.button} onPress={() => this.handleButtonPress()}>
                    <Text style={signUpStyles.text}>SignUp Screen</Text>
                </TouchableHighlight>
            </View>
        );
    }
}