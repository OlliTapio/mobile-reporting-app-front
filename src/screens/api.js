import { Platform, AsyncStorage, NetInfo } from 'react-native';

// Necessary because of a bug on iOS https://github.com/facebook/react-native/issues/8615#issuecomment-287977178
const isNetworkConnected = () => {
    if (Platform.OS === 'ios') {
        return new Promise(resolve => {
            const handleFirstConnectivityChangeIOS = isConnected => {
                NetInfo.isConnected.removeEventListener('connectionChange', handleFirstConnectivityChangeIOS);
                resolve(isConnected);
            };
            NetInfo.isConnected.addEventListener('connectionChange', handleFirstConnectivityChangeIOS);
        });
    }
    return NetInfo.isConnected.fetch();
};

/* Fetch data from ASyncStorage in case there is no internet connection.
   If no data has been stored an empty value will be returned. */
const fetchLocalData = (dataUrl) => {
    return AsyncStorage.getItem(dataUrl)
        .then(data => {
            if (data !== null) {
                return JSON.parse(data);
            } else {
                return [];
            }
        });
};

// Fetch remote data i.e. fetch data from the server.
const fetchRemoteData = (dataUrl) => {
    return (
        fetch(dataUrl)
            .then(response => {
                return response.json();
            })
    );
};

/* Store data to ASyncStorage, a simple key-value storage system global to the app.
   Keys and values are stored as a string. */
const saveData = (dataUrl, data) => {
    AsyncStorage.setItem(dataUrl, JSON.stringify(data));
};

/* Fetch data from the server or ASyncStorage, depending on the availability of internet connection */
export const fetchData = (dataUrl) => {
    return isNetworkConnected()
        .then((isConnected) => {
            if (!isConnected) { return fetchLocalData(dataUrl); }
            return fetchRemoteData(dataUrl);
        })
        .then((data) => {
            saveData(dataUrl, data);
            return data;
        });
};