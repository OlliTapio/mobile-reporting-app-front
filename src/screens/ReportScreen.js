import React from 'react';
import { View, ScrollView, TextInput, Alert, Text, ActivityIndicator, Linking } from 'react-native';
import { NavigationActions } from 'react-navigation';
import { Icon } from 'react-native-elements';
import ModalDropdown from 'react-native-modal-dropdown';
import moment from 'moment';
import { connect } from 'react-redux';
import { Checkbox } from '../components/Checkbox';
import { Radioform } from '../components/Radioform';
import { Dropdown } from '../components/Dropdown';
import { Button } from '../components/Button';
import { Datepicker } from '../components/Datepicker';
import { AppBackground } from '../components/AppBackground';
import { createNewReport, removeDraft, saveDraft } from './api';
import { strings } from '../locales/i18n';
import { insertFieldAnswer, emptyFields } from '../redux/actions/newReport';

import newReportStyles from './style/newReportStyles';
import templateScreenStyles from './style/templateScreenStyles';
import styles from '../components/Dropdown/styles';
import EStyleSheet from 'react-native-extended-stylesheet';


export class ReportScreen extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isDraft         : false, // for future use?
            report          : null,
            title           : null,
            isUnsaved       : true,
            isLoading       : true,
            number          : '',
            isEditable      : true,
            dataFieldsByID  : null,
        };
    }

    componentDidMount() {
        this.getFieldsByReportID();
    }

    getFieldsByReportID = () => {
        const { templateID, reportID } = this.props.navigation.state.params;
        const { reports } = this.props;

        const report = reports[templateID].find((obj) => obj.id === reportID);
        const fieldAnswers = report.answers;
        fieldAnswers.map((field) => this.props.dispatch(insertFieldAnswer(field, field.answer)));

        this.setState({ report: report });
        this.setState({ dataFieldsByID: fieldAnswers, isLoading: false });
    };

    // delete draft from asyncstorage
    deleteDraft = () => {
        const { templateID, reportID } = this.props.navigation.state.params;
        const { username } = this.props;

        removeDraft(username, templateID, reportID);
        this.props.dispatch(emptyFields());

        Alert.alert('Deleted draft.');

        this.props.navigation.state.params.refresh();
        this.props.navigation.dispatch(NavigationActions.back());
    };

    // save report locally in asyncstorage
    save = () => {
        const { username, answers } = this.props;
        const { templateID } = this.props.navigation.state.params;

        const report = this.state.report;
        report.title = this.state.title || report.title;
        report.dateCreated = moment().format('YYYY-MM-DD');
        report.answers = Object.values(answers);
        saveDraft(username, templateID, report);

        Alert.alert('Saved');

        this.props.dispatch(emptyFields());             // return newReport state to its initial state
        this.props.navigation.state.params.refresh();   // update templateScreen
        this.props.navigation.dispatch(NavigationActions.back());
    };

    // Inserts data to server with a post method.
    send = () => {
        const report = {
            templateID: this.props.navigation.state.params.templateID,
            title: this.props.navigation.state.params.title,
            dateCreated: moment().format('YYYY-MM-DD'),
            answers: [
                {
                    fieldID: 1,
                    answer: 'Answer 1'
                },
                {
                    fieldID: 2,
                    answer: 'Answer 2'
                }
            ]
        };

        createNewReport(this.props.username, report, this.props.token).then(response => {
            if (response.status === 200) {
                this.props.navigation.state.params.refresh();
                this.props.navigation.dispatch(NavigationActions.back());
                return strings('createNew.reportAdded');
            } else {
                return response.status;
            }
        }).then((message) => {
            // Showing response message coming from server after inserting records.
            Alert.alert(message);
        }).catch((error) => {
            console.error(error);
        });
    };

    render() {

        if (this.state.isLoading) {
            return (
                <AppBackground>
                    <ActivityIndicator
                        animating={this.state.animating}
                        style={[templateScreenStyles.activityIndicator, { height: 80 }]}
                        size='large'
                        color={'#9dcbe5'}
                    />
                </AppBackground>
            );
        }

        const { isEditable } = this.state;
        const { answers } = this.props;
        const renderedFields = (!this.state.dataFieldsByID) ? null : this.state.dataFieldsByID.map((field, index) => {

            switch (field.fieldID) {

                case 1: // Name TODO: all titles from db
                    return (
                        <View key={index} style={newReportStyles.fieldContainer}>
                            <Text style={newReportStyles.text}>Nimikenttä</Text>
                            <TextInput
                                editable={isEditable}
                                defaultValue={field.answer}
                                onChangeText={(text) => this.props.dispatch(insertFieldAnswer(field, text))}
                                underlineColorAndroid='transparent'
                                style={newReportStyles.textInput}
                            />
                        </View>
                    );

                case 2: // Checkbox
                    return (
                        <View key={index} style={newReportStyles.fieldContainer}>
                            <Text style={newReportStyles.text}>Checkbox</Text>
                            <Checkbox
                                key={index}
                                title={'This is a nice checkbox'}
                                editable={isEditable}
                                //The answer is saved as '1' or '0' but the component expects a boolean
                                isChecked={field.answer === '1'}
                                //The ability to dispatch the checkbox status is passed on to the component
                                //as a prop, and the component itself can call this function in its
                                //onPress, i.e. when the checkbox is pressed
                                onPressFunction={(answer) => this.props.dispatch(insertFieldAnswer(field, answer))}
                            />
                        </View>
                    );

                case 3: // Dropdown
                    return (
                        <View key={index} style={newReportStyles.fieldContainer}>
                            <Text style={ newReportStyles.text }>Dropdown</Text>
                            <ModalDropdown
                                disabled={!isEditable}
                                options={['option 1', 'option 2']}
                                dropdownStyle={styles.dropStyleClass}
                                style={styles.dropdownButton}
                                renderRow={ () =>
                                    <View>
                                        <ModalDropdown
                                            options={['option 3', 'option 4']}
                                            style={styles.lowerDropdownStyleClass}
                                            dropdownStyle={styles.dropStyleClass}
                                        />
                                    </View>
                                }
                            >
                                <View style={styles.buttonContent}>
                                    <Text style={styles.dropdownText}>
                                    Select option
                                    </Text>
                                    <Icon name={'expand-more'} color={EStyleSheet.value('$placeholder')} style={styles.icon}/>
                                </View>
                            </ModalDropdown>
                        </View>

                    );

                case 4: // TextRow (One row text field)
                    return (
                        <View key={index} style={newReportStyles.fieldContainer}>
                            <Text style={newReportStyles.text}>1-rivinen teksti</Text>
                            <TextInput
                                editable={isEditable}
                                defaultValue={field.answer}
                                underlineColorAndroid='transparent'
                                style={newReportStyles.textInput}
                                onChangeText={(text) => this.props.dispatch(insertFieldAnswer(field, text))}
                            />
                        </View>
                    );

                case 5: // Choice (Yes/No)
                    return (
                        <View key={index} style={newReportStyles.altFieldContainer}>
                            <Text style={newReportStyles.text}>Valintalaatikko (Kyllä/Ei)</Text>
                            <Radioform
                                options={[{ label: 'Yes', value: 0 }, { label: 'No', value: 1 }]}
                                editable={isEditable}
                                initial={JSON.parse(field.answer)}
                                onPress={(value) => this.props.dispatch(insertFieldAnswer(field, value))}
                            />
                        </View>
                    );

                case 6: // Calendar
                    return (
                        <View key={index} style={newReportStyles.fieldContainer}>
                            <Text style={newReportStyles.text}>Kalenteri</Text>
                            <Datepicker
                                editable={isEditable}
                                mode={'date'}
                                answer={answers[field.orderNumber] ? answers[field.orderNumber].answer : field.defaultValue}
                                onDateChange={(date) => this.props.dispatch(insertFieldAnswer(field, date))}
                            />
                        </View>
                    );

                case 7: // Instruction
                    return (
                        <View key={index} style={newReportStyles.fieldContainer}>
                            <Text style={newReportStyles.text}>Ohje</Text>
                            <Text style={newReportStyles.instructions}>
                                {field.defaultValue}
                            </Text>
                        </View>
                    );

                case 8: // Text (Multiple row text field)
                    return (
                        <View key={index} style={newReportStyles.fieldContainer}>
                            <Text style={ newReportStyles.text }>Laaja teksti</Text>
                            <TextInput
                                multiline
                                editable={isEditable}
                                style={ newReportStyles.multilineTextInput }
                                onChangeText = {(text) => this.props.dispatch(insertFieldAnswer(field, text))}
                                placeholder = {field.answer}
                                defaultValue={field.answer}
                                placeholderTextColor={EStyleSheet.value('$placeholder')}
                            />
                        </View>
                    );

                case 9: // Time
                    return (
                        <View key={index} style={newReportStyles.fieldContainer}>
                            <Text style={ newReportStyles.text }>Kellonaika</Text>
                            <Datepicker
                                editable={isEditable}
                                mode={'time'}
                                answer={answers[field.orderNumber] ? answers[field.orderNumber].answer : field.defaultValue}
                                onDateChange = {(time) => this.props.dispatch(insertFieldAnswer(field, time))}
                            />
                        </View>
                    );

                case 10: // Digits (Text input that only accepts numeric characters)
                    return (
                        <View key={index} style={newReportStyles.fieldContainer}>
                            <Text style={ newReportStyles.text }>Numerokenttä</Text>
                            <TextInput
                                editable={isEditable}
                                style={ newReportStyles.textInput }
                                defaultValue={field.answer}
                                keyboardType = 'numeric'
                                onChangeText={(text) => this.props.dispatch(insertFieldAnswer(field, text))}
                            />
                        </View>
                    );

                case 11: // Link
                    return (
                        <View key={index} style={newReportStyles.fieldContainer}>
                            <Text style={newReportStyles.text}>Linkkikenttä</Text>
                            <View style={newReportStyles.linkContainer}>
                                <Icon name={'link'} type={'feather'} iconStyle={ newReportStyles.linkIcon }/>
                                <Text
                                    disabled={!isEditable}
                                    style={ newReportStyles.link }
                                    onPress={() => Linking.openURL(field.answer)}>
                                    Link to somewhere
                                </Text>
                            </View>
                        </View>
                    );

                case 12: // User dropdown
                    return (
                        <View key={index} style={newReportStyles.fieldContainer}>
                            <Text style={ newReportStyles.text }>Käyttäjät</Text>
                            <Dropdown
                                ref={ ModalDrop => this.modalDropdown = ModalDrop }
                                disabled={!isEditable}
                                defaultValue={field.answer}
                                options={field.answer.split(',')}/>
                        </View>
                    );

                default:
                    return (
                        null
                    );

            }
        });

        return (
            <AppBackground>
                <View style={newReportStyles.ViewContainer}>
                    <ScrollView keyboardShouldPersistTaps={'handled'} style={newReportStyles.ReportScrollView}>
                        {renderedFields}
                        {
                            (this.props.navigation.state.params.reportID < 0) &&
                            <View>
                                <Button title={strings('createNew.save')} key={999} type={'save'} onPress={ () => this.save()} />
                                <Button title={strings('createNew.send')} type={'send'} onPress={() => console.log('send')}  />
                                <Button title={'Delete'} type={'delete'} onPress={() => this.deleteDraft()} />
                            </View>
                        }
                    </ScrollView>
                </View>

            </AppBackground>
            /*
                <TextInput
                    placeholder={ strings('createNew.enterNew') }
                    onChangeText={(TextInputName) => this.setState({ TextInputName })}
                    underlineColorAndroid='transparent'
                    style={newReportStyles.textInput}
                />
                <Button
                    title={ strings('createNew.createNew') }
                    onPress={ this.send }
                >
                </Button>
            */
        );
    }
}


// maps redux state to component props. Object that is returned can be accessed via 'this.props' e.g. this.props.email
const mapStateToProps = (state) => {
    const token         = state.user.token;
    const username      = state.user.username;
    const number        = state.newReport.number;
    const answers       = state.newReport.answers;
    const reports       = state.reports;
    return {
        username,
        number,
        token,
        reports,
        answers
    };
};

export default connect(mapStateToProps)(ReportScreen);
