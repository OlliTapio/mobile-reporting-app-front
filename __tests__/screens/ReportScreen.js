import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'isomorphic-fetch';
//import { shallow } from 'enzyme';
//import { ActivityIndicator } from 'react-native';

import { ReportScreen } from '../../src/screens/ReportScreen';
//import { url } from '../../src/screens/urlsetting';

configure({ adapter: new Adapter() });

const navigation = { state: { params: { templateID: 1, reportID: 1, title: 'Testiraportti' } } };
const templates = {
    1: {
        template_id: 1,
        title: 'Template 1',
        fields: [
            {
                field_id: 1,
                template_id: 1,
                order_number: 1,
                title: 'Name',
                required: true,
                type: 'TEXTFIELD_SHORT',
                default_value: null,
                field_options: null
            },
            {
                field_id: 2,
                template_id: 1,
                order_number: 2,
                title: 'Options',
                required: true,
                type: 'RADIOBUTTON',
                default_value: null,
                field_options: [
                    {
                        field_option_id: 3,
                        field_id: 2,
                        value: 'Option 1',
                        default_value: true
                    },
                    {
                        field_option_id: 4,
                        field_id: 2,
                        value: 'Option 2',
                        default_value: false
                    }
                ]
            }
        ]
    }
};

const reports = {
    '1': [{
        report_id: 1,
        user_id: 1,
        template_id: 1,
        title: 'Testiraportti',
        date_created: '2018-03-16',
        date_accepted: null,
        string_answers: [
            {
                string_answer_id: 208,
                report_id: 1,
                field_id: 1,
                value: 'Test1'
            }
        ],
        option_answers: [
            {
                option_answer_id: null,
                report_id: 1,
                field_option_id: 3,
                selected: true,
            },
            {
                option_answer_id: null,
                report_id: 1,
                field_option_id: 4,
                selected: false
            }
        ]
    }]
};

const report = {
    report_id: 1,
    user_id: 1,
    template_id: 1,
    title: 'Testiraportti',
    date_created: '2018-03-16',
    date_accepted: null,
    string_answers: [
        {
            string_answer_id: 208,
            report_id: 1,
            field_id: 1,
            value: 'Test1'
        }
    ],
    option_answers: [
        {
            option_answer_id: null,
            report_id: 1,
            field_option_id: 3,
            selected: true,
        },
        {
            option_answer_id: null,
            report_id: 1,
            field_option_id: 4,
            selected: false
        }
    ]
};

it('renders correctly', () => {
    const tree = renderer.create(
        <ReportScreen
            navigation={navigation}
            reports={reports}
            report={report}
            templates={templates}
            dispatch={jest.fn()}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});