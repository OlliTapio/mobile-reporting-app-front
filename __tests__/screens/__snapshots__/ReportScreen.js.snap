// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  colors={
    Array [
      undefined,
      undefined,
      undefined,
    ]
  }
  style={undefined}
>
  <View
    style={undefined}
  >
    <RCTScrollView
      keyboardShouldPersistTaps="handled"
      style={undefined}
    >
      <View>
        <View
          style={undefined}
        >
          <View
            style={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={undefined}
            >
              Otsikko
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={undefined}
            >
               *
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            defaultValue="Testiraportti"
            editable={false}
            onChangeText={[Function]}
            placeholder={null}
            placeholderTextColor={null}
            style={undefined}
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          pointerEvents="none"
        >
          <View
            style={undefined}
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={undefined}
              >
                Name
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={undefined}
              >
                 *
              </Text>
            </View>
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder={null}
              placeholderTextColor={null}
              style={undefined}
              underlineColorAndroid="transparent"
              value="Test1"
            />
          </View>
          <View
            style={undefined}
          >
            <View
              style={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={undefined}
              >
                Options
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={undefined}
              >
                 *
              </Text>
            </View>
            <View
              style={
                Array [
                  undefined,
                  undefined,
                  false,
                ]
              }
            >
              <View
                style={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "flexDirection": "row",
                        "marginBottom": 5,
                      },
                      Array [
                        undefined,
                        undefined,
                      ],
                      false,
                    ]
                  }
                >
                  <View
                    style={undefined}
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "borderColor": "#2196f3",
                            "borderRadius": 30,
                            "height": 30,
                            "justifyContent": "center",
                            "width": 30,
                          },
                          Object {
                            "borderColor": "#2196f3",
                            "borderWidth": 1.5,
                          },
                          undefined,
                          Object {
                            "borderRadius": 12,
                            "height": 24,
                            "width": 24,
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "borderRadius": 10,
                            },
                            false,
                            false,
                            false,
                          ]
                        }
                      />
                    </View>
                  </View>
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    hitSlop={undefined}
                    nativeID={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        undefined,
                        Object {
                          "alignItems": "center",
                          "alignSelf": "center",
                          "flexDirection": "row",
                        },
                      ]
                    }
                    testID={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "lineHeight": 20,
                            "paddingLeft": 10,
                          },
                          false,
                          Object {
                            "color": undefined,
                          },
                          undefined,
                        ]
                      }
                    >
                      Option 1
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "flexDirection": "row",
                        "marginBottom": 5,
                      },
                      Array [
                        undefined,
                        undefined,
                      ],
                      false,
                    ]
                  }
                >
                  <View
                    style={undefined}
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "borderColor": "#2196f3",
                            "borderRadius": 30,
                            "height": 30,
                            "justifyContent": "center",
                            "width": 30,
                          },
                          Object {
                            "borderColor": "#2196f3",
                            "borderWidth": 1.5,
                          },
                          undefined,
                          Object {
                            "borderRadius": 12,
                            "height": 24,
                            "width": 24,
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "borderRadius": 10,
                            },
                            false,
                            false,
                            false,
                          ]
                        }
                      />
                    </View>
                  </View>
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    hitSlop={undefined}
                    nativeID={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Array [
                        undefined,
                        Object {
                          "alignItems": "center",
                          "alignSelf": "center",
                          "flexDirection": "row",
                        },
                      ]
                    }
                    testID={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "lineHeight": 20,
                            "paddingLeft": 10,
                          },
                          false,
                          Object {
                            "color": undefined,
                          },
                          undefined,
                        ]
                      }
                    >
                      Option 2
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
